src/test/java/tests/theInternetHerokuappTests/DynamicControls.java

```diff
@@ -0,0 +1,40 @@
+package tests.theInternetHerokuappTests;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import tests.BaseTest;
+
+public class DynamicControls extends BaseTest {
+
+    @Test
+    public void addRemoveElementsTest() {
+        driver.get("http://the-internet.herokuapp.com/dynamic_controls");
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        WebElement checkbox = driver.findElement(By.cssSelector("[type=checkbox]"));
+        checkbox.click();
+        WebElement removeAddButton = driver.findElement(By.xpath("//*[@id='checkbox-example']/descendant::button[@type='button']"));
+        removeAddButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("message")));
+        wait.until(ExpectedConditions.invisibilityOf(checkbox));
+        removeAddButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("message")));
```

Comment:
Either use By.xpath(contains text()) locator to  wait for a message with particular text

OR

wait.until(ExpectedConditions.textToBe(By.cssSelector("#message", "it'g gone / it's back")));)